[Bug]: fails to build using Dockerfile due to node version missmatch

### Guidelines

- [x] I have encountered this bug in the [latest release of FreeTube](https://github.com/FreeTubeApp/FreeTube/releases).
- [x] I have searched the issue tracker for [open](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aopen+is%3Aissue) and [closed](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aissue+is%3Aclosed) issues that are similar to the bug report I want to file, without success.
- [x] I have searched the [documentation](https://docs.freetubeapp.io/) for information that matches the description of the bug I want to file, without success.
- [x] This issue contains only one bug.

### Describe the bug

1. `git clone https://github.com/MarmadileManteater/FreeTubeAndroid`
2. run `docker build -t freetubecordova .` as instructed in the README
3. docker fails to build with the following error message:
```bash
docker build -t freetubecordova . 
[+] Building 156.0s (14/18)                                                                                                                   docker:default 
 => [internal] load build definition from Dockerfile                                                                                                    0.0s 
 => => transferring dockerfile: 698B                                                                                                                    0.0s 
 => WARN: FromAsCasing: 'as' and 'FROM' keywords' casing do not match (line 24)                                                                         0.0s 
 => [internal] load metadata for docker.io/library/nginx:latest                                                                                         1.6s 
 => [internal] load metadata for docker.io/library/node:18-alpine                                                                                       1.5s 
 => [internal] load .dockerignore                                                                                                                       0.0s 
 => => transferring context: 92B                                                                                                                        0.0s 
 => [app 1/2] FROM docker.io/library/nginx:latest@sha256:6784fb0834aa7dbbe12e3d7471e69c290df3e6ba810dc38b34ae33d3c1c05f7d                               4.4s 
 => => resolve docker.io/library/nginx:latest@sha256:6784fb0834aa7dbbe12e3d7471e69c290df3e6ba810dc38b34ae33d3c1c05f7d                                   0.0s 
 => => sha256:6784fb0834aa7dbbe12e3d7471e69c290df3e6ba810dc38b34ae33d3c1c05f7d 10.25kB / 10.25kB                                                        0.0s 
 => => sha256:fafd5f55296511cd0f1c991efa0cb85f323449bf643f3b6d760c6542f020d6e5 2.29kB / 2.29kB                                                          0.0s 
 => => sha256:1e5f3c5b981a9f91ca91cf13ce87c2eedfc7a083f4f279552084dd08fc477512 8.58kB / 8.58kB                                                          0.0s 
 => => sha256:3b00567da96412a0298328fad6b96aadd3bc9ca97e6683534da152b0b9e8a977 44.15MB / 44.15MB                                                        1.2s 
 => => sha256:56b81cfa547d78176e719c9cd387e35fbd95ce6e8500efb26d9fbcbac4dbc4f6 628B / 628B                                                              0.5s 
 => => sha256:1bc5dc8b475d9de62c94cf69a139df17aaf7fe7bac76b5133d8048a058fb3c3b 956B / 956B                                                              0.2s 
 => => sha256:979e6233a40af9a51ee2c9894bd42ebf78a79c87cbed70cc6a3840a0952f3d57 405B / 405B                                                              0.5s 
 => => sha256:d2a7ba8dbfee4f222961ad1449648ab46a6cea044e4cfb374450325ecee8f67f 1.21kB / 1.21kB                                                          0.7s 
 => => sha256:32e44235e1d55a34be577baa12271f067c0bf79ba4e1ca4e6ec484424540132a 1.40kB / 1.40kB                                                          0.7s 
 => => extracting sha256:3b00567da96412a0298328fad6b96aadd3bc9ca97e6683534da152b0b9e8a977                                                               2.0s 
 => => extracting sha256:56b81cfa547d78176e719c9cd387e35fbd95ce6e8500efb26d9fbcbac4dbc4f6                                                               0.0s 
 => => extracting sha256:1bc5dc8b475d9de62c94cf69a139df17aaf7fe7bac76b5133d8048a058fb3c3b                                                               0.0s 
 => => extracting sha256:979e6233a40af9a51ee2c9894bd42ebf78a79c87cbed70cc6a3840a0952f3d57                                                               0.0s 
 => => extracting sha256:d2a7ba8dbfee4f222961ad1449648ab46a6cea044e4cfb374450325ecee8f67f                                                               0.0s 
 => => extracting sha256:32e44235e1d55a34be577baa12271f067c0bf79ba4e1ca4e6ec484424540132a                                                               0.0s 
 => [dep 1/7] FROM docker.io/library/node:18-alpine@sha256:8d6421d663b4c28fd3ebc498332f249011d118945588d0a35cb9bc4b8ca09d9e                             4.5s 
 => => resolve docker.io/library/node:18-alpine@sha256:8d6421d663b4c28fd3ebc498332f249011d118945588d0a35cb9bc4b8ca09d9e                                 0.0s 
 => => sha256:ee77c6cd7c1886ecc802ad6cedef3a8ec1ea27d1fb96162bf03dd3710839b8da 6.18kB / 6.18kB                                                          0.0s 
 => => sha256:8d6421d663b4c28fd3ebc498332f249011d118945588d0a35cb9bc4b8ca09d9e 7.67kB / 7.67kB                                                          0.0s 
 => => sha256:929b04d7c782f04f615cf785488fed452b6569f87c73ff666ad553a7554f0006 1.72kB / 1.72kB                                                          0.0s 
 => => sha256:dd71dde834b5c203d162902e6b8994cb2309ae049a0eabc4efea161b2b5a3d0e 40.01MB / 40.01MB                                                        2.2s 
 => => sha256:1e5a4c89cee5c0826c540ab06d4b6b491c96eda01837f430bd47f0d26702d6e3 1.26MB / 1.26MB                                                          1.2s 
 => => sha256:25ff2da83641908f65c3a74d80409d6b1b62ccfaab220b9ea70b80df5a2e0549 446B / 446B                                                              1.4s 
 => => extracting sha256:dd71dde834b5c203d162902e6b8994cb2309ae049a0eabc4efea161b2b5a3d0e                                                               1.9s 
 => => extracting sha256:1e5a4c89cee5c0826c540ab06d4b6b491c96eda01837f430bd47f0d26702d6e3                                                               0.2s 
 => => extracting sha256:25ff2da83641908f65c3a74d80409d6b1b62ccfaab220b9ea70b80df5a2e0549                                                               0.0s 
 => [internal] load build context                                                                                                                       0.5s 
 => => transferring context: 51.63MB                                                                                                                    0.4s 
 => [dep 2/7] WORKDIR /app                                                                                                                              0.0s 
 => [build 3/6] COPY . .                                                                                                                                0.3s 
 => [dep 3/7] COPY package.json ./package.json                                                                                                          0.1s 
 => [dep 4/7] COPY yarn.lock ./yarn.lock                                                                                                                0.2s 
 => [dep 5/7] COPY dis[t]/web ./dist/web                                                                                                                0.0s 
 => [dep 6/7] RUN apk add git                                                                                                                           3.2s 
 => ERROR [dep 7/7] RUN if [ ! -d 'dist/web' ]; then yarn ci; fi                                                                                      146.2s 
------ 
 > [dep 7/7] RUN if [ ! -d 'dist/web' ]; then yarn ci; fi: 
0.559 yarn run v1.22.22 
0.611 $ yarn install --silent --frozen-lockfile --network-concurrency 1 
145.0 warning autolinker@4.1.5: The engine "pnpm" appears to be invalid. 
145.1 error minimatch@10.0.1: The engine "node" is incompatible with this module. Expected version "20 || >=22". Got "18.20.8" 
145.1 error Found incompatible module. 
145.2 error Command failed with exit code 1. 
145.2 info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command. 
------ 
 1 warning found (use docker --debug to expand): 
 - FromAsCasing: 'as' and 'FROM' keywords' casing do not match (line 24) 
Dockerfile:12 
-------------------- 
  10 |     RUN apk add git 
  11 |     # don't rebuild if you don't have to 
  12 | >>> RUN if [ ! -d 'dist/web' ]; then yarn ci; fi 
  13 | 
  14 |     ## Build Stage ## 
-------------------- 
ERROR: failed to solve: process "/bin/sh -c if [ ! -d 'dist/web' ]; then yarn ci; fi" did not complete successfully: exit code: 1
```

### Expected Behavior

docker should build the project just fine

### Issue Labels

feature stopped working

### FreeTube Version

42c08316af364167436b6d7259bcf624ab5a1522

### Operating System Version

Arch Linux (6.14.2-arch1-1)

### Installation Method

PWA (Progressive Web App)

### Primary API used

Local API

### Last Known Working FreeTube Version (If Any)

_No response_

### Additional Information

`minimatch` dependency supposedly requires node version >=20:
```bash
error minimatch@10.0.1: The engine "node" is incompatible with this module. Expected version "20 || >=22". Got "18.20.8" 
```

maybe bump node to v22 in the Dockerfile (`node:22-alpine`)?

### Nightly Build

- [ ] I have encountered this bug in the latest [nightly build](https://docs.freetubeapp.io/development/nightly-builds).

update:

I took the liberty of modifying the Dockerfile and changing `node:18-alpine` to `node:22-alpine` and it then failed again on:
```Dockerfile
RUN if [ ! -d 'dist/web' ]; then yarn ci; fi
```
this time on the `postinstall` phase of `yarn ci` and it was complaining about
```bash
Error: Cannot find module '/app/_scripts/patchShaka.mjs'
182.2     at Function._resolveFilename (node:internal/modules/cjs/loader:1401:15)
182.2     at defaultResolveImpl (node:internal/modules/cjs/loader:1057:19)
182.2     at resolveForCJSWithHooks (node:internal/modules/cjs/loader:1062:22)
182.2     at Function._load (node:internal/modules/cjs/loader:1211:37)
182.2     at TracingChannel.traceSync (node:diagnostics_channel:322:14)
182.2     at wrapModuleLoad (node:internal/modules/cjs/loader:235:24)
182.2     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:171:5)
182.2     at node:internal/main/run_main_module:36:49 {
182.2   code: 'MODULE_NOT_FOUND',
182.2   requireStack: []
182.2 }
```

so I added
```Dockerfile
COPY _scripts ./_scripts
```

and then it successfully built the image.

I tested it a bit and everything seemed to work fine after 👍

However, I'm not familiar with the project's update and build pipelines, so I'm not entirely sure if introducing these changes would lead to any unintended side effects. In any case, here is my final `Dockerfile`:
```Dockerfile

## Dependency Stage ##
FROM node:22-alpine AS dep
WORKDIR /app
COPY package.json ./package.json
COPY yarn.lock ./yarn.lock
COPY _scripts ./_scripts
# copy `dist` if it exists already
COPY dis[t]/web ./dist/web
# git is needed for jinter
RUN apk add git
# don't rebuild if you don't have to
RUN if [ ! -d 'dist/web' ]; then yarn ci; fi

## Build Stage ##
FROM node:22-alpine AS build
WORKDIR /app
COPY . .
COPY --from=dep /app/dis[t]/web ./dist/web
COPY --from=dep /app/node_module[s] ./node_modules
# don't rebuild if you don't have to
RUN if [ ! -d 'dist/web' ]; then yarn pack:web; fi

## App Stage ##
FROM nginx:latest as app
COPY --from=build /app/dist/web /usr/share/nginx/html
```

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,9 +1,10 @@
 
 ## Dependency Stage ##
-FROM node:18-alpine AS dep
+FROM node:22-alpine AS dep
 WORKDIR /app
 COPY package.json ./package.json
 COPY yarn.lock ./yarn.lock
+COPY _scripts ./_scripts
 # copy `dist` if it exists already
 COPY dis[t]/web ./dist/web
 # git is needed for jinter
@@ -12,7 +13,7 @@ RUN apk add git
 RUN if [ ! -d 'dist/web' ]; then yarn ci; fi
 
 ## Build Stage ##
-FROM node:18-alpine AS build
+FROM node:22-alpine AS build
 WORKDIR /app
 COPY . .
 COPY --from=dep /app/dis[t]/web ./dist/web

